test/test_datadog_agent.cpp

```diff
@@ -101,7 +118,7 @@ TEST_CASE("CollectorResponse") {
     // Don't echo error messages.
     logger->echo = nullptr;
 
-    auto status = GENERATE(range(300, 600));
+    auto status = GENERATE(range(201, 600));
```

Comment:
Consider adding a comment here saying why `[201, 299]` are considered errors, even though HTTP says otherwise.

One way to say it is that we know that the Datadog Agent in particular only ever returns 200 on success, and never any of the other 2xx codes. Also there's the corner case where 200 + empty body can mean 429 if the agent is configured in a certain way.

Reply:
Added a comment in 8c405b49c68a5c20b5417a1c76261e1e49672b75.